### PR TITLE
feat(python): More Python AST Nodes

### PR DESCRIPTION
**Pull request by noanflaherty**
This PR introduces the following new ATS nodes for python. Most are directly influenced from csharp.

* `CodeBlock` – I also refactored existing initializers to use these instead of strings
* `Method` – Feels more Pythonic to call these "Functions," but I stuck to "Method" for consistency
* `Parameter` – Used to define arguments on a Method. At first my instinct was to reuse `Field` for this, but opted for consistency
* `MethodInvocation` – Used to invoke a method
* `MethodArgument` – Passed to MethodInvocation and a departure from csharp. Needed by python to support positional vs keyword arguments.

It also adds support for defining a class's parent class(es) and generic types.

## Summary by CodeRabbit

- **New Features**
	- Enhanced `.gitignore` to exclude additional workspace configuration files.
	- Added new functionalities for class handling, including inheritance and field management in the Python code generation module.
	- Introduced a structured way to define and generate methods, including support for static, instance, and class methods.

- **Bug Fixes**
	- Improved handling of parent class references and field management in class definitions.

- **Tests**
	- Expanded test coverage for class definitions, inheritance scenarios, and various field configurations.
	- Added comprehensive unit tests for method generation and representation.

- **Documentation**
	- Updated export structure to include new entities for better accessibility.

### CODE REVIEW

File: generators/pythonv2/codegen/src/ast/Class.ts [hunk 1]
```diff
@@ -6,21 +7,38 @@ export declare namespace Class {
     interface Args {
         /* The name of the Python class */
         name: string;
+        /* The parent classes that this class inherits from */
+        parentClassReferences?: ClassReference[];
```
Comment (dsinghvi) [inline generators/pythonv2/codegen/src/ast/Class.ts:11]: nit: would name this `extends`

File: generators/pythonv2/codegen/src/ast/Method.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        /* The body of the method */
+        body?: CodeBlock;
+        /* The docstring for the method */
+        docstring?: string;
+        /* The type of the method, defaults to STATIC */
+        type?: MethodType;
+        /* The class this method belongs to, if any */
+        classReference?: ClassReference;
+        /* Any decorators to add to the method */
+        decorators?: string[];
```
Comment (dsinghvi) [inline generators/pythonv2/codegen/src/ast/Method.ts:32]: do decorators need to be imported often? I could see this being a `ClassReference`

File: generators/pythonv2/codegen/src/ast/MethodArgument.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { CodeBlock } from "./CodeBlock";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { Type } from "./Type";
+
+export declare namespace MethodArgument {
```
Comment (dsinghvi) [inline generators/pythonv2/codegen/src/ast/MethodArgument.ts:6]: nit: we typically call this `Parameter`

File: generators/pythonv2/codegen/src/ast/MethodArgument.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    interface Args {
+        /* If a kwarg, then the name of the parameter that this is a keyword argument for */
+        name?: string;
+        /* The value of the argument */
+        value: CodeBlock;
+    }
```
Comment (dsinghvi) [inline generators/pythonv2/codegen/src/ast/MethodArgument.ts:12]: whats the difference between `Parameter` and `MethodArgument` ?

Comment (dsinghvi) [reply generators/pythonv2/codegen/src/ast/MethodArgument.ts:12]: makes sense

**Review by dsinghvi — APPROVED** [review]

File: generators/pythonv2/codegen/src/ast/Method.ts (hunk 2 above)
Comment (noanflaherty) [reply generators/pythonv2/codegen/src/ast/Method.ts:32]: I'm going to remove decorators from the scope of this PR and add them in separately and more hollistically.

I'm thinking about adding a new top-level `Decorator` ast node and adding `addDecorator` methods to both the `Class` and `Method` ast nodes.

**Pull request merged**